### DISCUSSION TITLE
Bugfix - elastalert.util.format_index

### DISCUSSION THREAD

**Discussion by jmacdone** [jertel/elastalert2#1326] [Q&A] [answered]
I didn't tick all the boxes for a proper PR, but see: https://github.com/jmacdone/elastalert2/commit/38b7c1f41d60201764e3c575437f7abbc788ac1d

In short, I'm using `logstash-%Y.%m.%d.%H` (i.e. hourly indexes), but the format_index is only tuned for daily indexes (the default for logstash).  This checks for `%H` in the index pattern and changes to using a `hours=1` timedetla.  Better way?

**Comment by nsano-rururu** [comment]
It's an issue, not a pull request. Please submit a pull request.

**Comment by jertel** [comment]
I don't see an issue with this approach if you do want to submit a proper PR with all the requirements. I'm converting this issue into a Discussion since it doesn't represent a bug in the project, but more represents a discussion around a potential enhancement.

**Comment by jmacdone** [comment, marked as answer]
I'll submit a PR*.  It look like the pattern is to open a pull request, then add a commit to update CHANGELOG.md with whatever the URL for the PR ends up being.  Is that correct?

*And, yes, it did need unit tests.  The first try "works on my machine" but failed the existing unit tests with some off-by-one errors.

**Reply by jertel** [reply]
Yes, that's the correct PR process.
